test/dynamo/test_export_mutations.py

```diff
@@ -128,6 +92,44 @@ def forward(self, x):
         opt_mod = torch._dynamo.optimize("eager", nopython=True)(mod)
         self.assertTrue(torch._dynamo.utils.same(opt_mod(arg), real_result))
 
+    def test_module_attribute_mutation_violation_negative_5(self):
+        # Mutating attribute with a scalar type
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = 2
+
+            def forward(self, x):
+                self.a = self.a * 3
+                return x.sum() + self.a
+
+        self.check_same_with_export(Foo(), torch.ones(3, 2))
+
+    def test_module_attribute_mutation_violation_negative_6(self):
+        # Setting a new attribute inside forward()
+        class Foo(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.a = torch.Tensor(3, 2)
+
+            def forward(self, x):
+                self.b = 2
+                return x.sum() + self.a.sum() + self.b
+
+        self.check_same_with_export(Foo(), torch.Tensor(3, 2))
+
+    def test_module_attribute_mutation_violation_negative_7(self):
```

Comment:
while we're at it, I'd really love if we can properly name these tests instead of using numbers. All of these share `module_attribute_mutation_violation` so you can simply shorten the names to begin with, and then use some descriptive suffixes (maybe keeping positive / negative, because that's kind of useful...or, if you wish, success / failure).

Reply:
Hahah sounds good